packages/app/src/systems/Account/components/AccountList/AccountList.tsx

```diff
@@ -19,17 +20,20 @@ export function AccountList({
   onPress,
   onExport,
   onUpdate,
+  onToggleHidden,
 }: AccountListProps) {
   const [showHidden, setShowHidden] = useState(() => false);
   const [anyHiddenAccounts, setAnyHiddenAccounts] = useState(false);
+  const [isPrimaryAccount, setIsPrimaryAccount] = useState(false);
 
   function toggle() {
     setShowHidden((s) => !s);
   }
 
   useEffect(() => {
-    const hiddenAccounts = (accounts ?? []).some((acc) => acc.isHidden);
-    setAnyHiddenAccounts(hiddenAccounts);
+    const hiddenAccounts = (accounts ?? []).filter((acc) => acc.isHidden);
+    setAnyHiddenAccounts(hiddenAccounts.length > 0);
+    setIsPrimaryAccount((accounts ?? []).length - hiddenAccounts.length === 1);
```

Comment:
both hiddenAccounts and PrimaryAccount logics could be implemented in `useAccounts` hook

I would create selectors like:
```
  hasHiddenAccounts(state: AccountsMachineState) {
    return (state.context?.accounts?.some((acc) => acc.isHidden));
  },
  canHideAccounts(state: AccountsMachineState) {
    return (state.context?.accounts?.filter((acc) => !acc.isHidden)?.length || 0) > 1;
  },
```

and export data from the hook... then can remove this whole `useEffect` `useState` `setAnyHiddenAccounts` ... logic

Reply:
Thanks for pointing this out 🙏